### PR TITLE
RatingStatistics UI for management page added, exporter to Excel file implemented

### PR DESCRIPTION
**Pull request by TarasDovganyuk**

### CODE REVIEW

**Review by Yura09 — APPROVED** [review]

File: core/src/main/java/greencity/dto/ratingstatistics/RatingStatisticsDto.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Getter
+@Setter
+@EqualsAndHashCode
+public class RatingStatisticsDto {
+    private Long id;
+    private ZonedDateTime createDate;
+    private RatingCalculationEnum ratingCalculationEnum;
+    private float pointsChanged;
+    private float rating;
+    private User user;
```
Comment (antonbakalets) [inline core/src/main/java/greencity/dto/ratingstatistics/RatingStatisticsDto.java:19]: There will be a problem later. `RatingStatisticsDto` is a DTO from Controllers layer. `User` is an Entity from DAO layer. Please either replace by something like `String username` or introduce a kind of `UserDto`.

File: core/src/main/java/greencity/exporter/RatingExcelExporter.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private List<RatingStatisticsDto> ratingStatisticsDtos;
+
+    /**
+     * Constructor.
+     *
+     * @author Dovganyuk Taras
+     */
+    public RatingExcelExporter(List<RatingStatisticsDto> ratingStatisticsDtos) {
+        this.ratingStatisticsDtos = ratingStatisticsDtos;
+        workbook = new XSSFWorkbook();
```
Comment (antonbakalets) [inline core/src/main/java/greencity/exporter/RatingExcelExporter.java:32]: This design is not thread save. Imagine the `export()` method is invoked from another thread.

File: core/src/main/java/greencity/exporter/RatingExcelExporter.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * @author Dovganyuk Taras
+     */
+    public void export(HttpServletResponse response) throws IOException {
+        writeHeaderRow();
+        writeDataRows();
+
+        ServletOutputStream outputStream = response.getOutputStream();
+        workbook.write(outputStream);
+        workbook.close();
```
Comment (antonbakalets) [inline core/src/main/java/greencity/exporter/RatingExcelExporter.java:121]: `workbook` implements `AutoClosable`, so it should be closed in a `try-with-resource` block.

File: core/src/main/java/greencity/repository/RatingStatisticsRepo.java [hunk 4]
```diff
@@ -19,4 +23,21 @@
...
+     */
+    Page<RatingStatistics> findAll(Pageable pageable);
+
+    /**
+     * Find all {@link RatingStatistics}.
+     *
+     * @return {@link RatingStatistics};
+     * @author Dovganyuk Taras
+     */
+    List<RatingStatistics> findAll();
```
Comment (antonbakalets) [inline core/src/main/java/greencity/repository/RatingStatisticsRepo.java:42]: Method `findAll()` is already available in `JpaRepository` that this interface extends, so no need to introduce it here.

File: core/src/main/java/greencity/exporter/RatingExcelExporter.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            sheet.autoSizeColumn(6);
+        }
+    }
+
+    /**
+     * Export {@link greencity.entity.RatingStatistics} data to Excel file.
+     *
+     * @author Dovganyuk Taras
+     */
+    public void export(HttpServletResponse response) throws IOException {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/exporter/RatingExcelExporter.java:115]: Please pass as parameter only the object you really need. You need only the `OutputStream`, not the `HttpServletResponse`.

File: core/src/main/java/greencity/webcontroller/ManagementRatingStatisticsController.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+     * Export {@link RatingStatistics} to Excel file.
+     *
+     * @author Dovganyuk Taras
+     */
+    @GetMapping("/export")
+    public void exportToExcel(HttpServletResponse response) throws IOException {
+        response.setContentType("application/octet-stream");
+        String headerKey = "Content-Disposition";
+
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (antonbakalets) [inline core/src/main/java/greencity/webcontroller/ManagementRatingStatisticsController.java:62]: `dateFormat` can be created once, as a constant or a Singleton bean.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: core/src/main/java/greencity/exporter/RatingExcelExporter.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+                sheet.autoSizeColumn(5);
+
+                cell = row.createCell(6);
+                cell.setCellValue(dto.getRating());
+                sheet.autoSizeColumn(6);
+            }
+
+            workbook.write(outputStream);
+        } catch (IOException ex) {
+            ex.printStackTrace();
```
Comment (antonbakalets) [inline core/src/main/java/greencity/exporter/RatingExcelExporter.java:101]: Don't use that in real life. Replace by

`logger.error("Some Message", ex);`

**Review by antonbakalets — CHANGES_REQUESTED** [review]

**Review by antonbakalets — APPROVED** [review]

**Review by eyouu — APPROVED** [review]

**Review by forestik — APPROVED** [review]

**Pull request merged**